docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/DockerContainerDefinitionParser.java

```diff
@@ -98,36 +101,113 @@ public static DockerCompositions convert(String content, DefinitionFormat defini
             return new DockerCompositions();
         }
-        return convert(Paths.get(defaultUri), definitionFormat);
+        return convert(defaultUri, definitionFormat);
     }
+
+    private static URI getDefaultFileLocation(String filename) {
+
+        // src/{test/main}/docker
+        URI docker = checkSrcTestAndMainResources(filename, "docker");
+
+        if (docker == null) {
+
+            // .
+            docker = checkRoot(filename);
+
+            if (docker == null) {
+                // src/distribution
+                docker = checkDistributionDirectory(filename);
+
+                if (docker == null) {
+                    // src/{test, main}/resources/docker
+                    docker = checkSrcTestAndMainResources(filename, "resources/docker");
+
+                    if (docker == null) {
+
+                        // src/{test, main}/resources
+                        docker = checkSrcTestAndMainResources(filename, "resources");
+                    }
+                }
+            }
+        }
+
+        return docker;
+
+    }
+
+    private static URI checkRoot(String filename) {
+        final Path rootPath = Paths.get(filename);
+        Path finalPath;
+
+        if ((finalPath = exists(rootPath)) != null) {
+            return finalPath.toUri();
+        }
+
+        return null;
+    }
+
+    private static URI checkDistributionDirectory(String filename) {
+        final Path rootPath = Paths.get("src", "distribution", filename);
+        Path finalPath;
+
+        if ((finalPath = exists(rootPath)) != null) {
+            return finalPath.toUri();
+        }
+
+        return null;
+
+    }
+
+    /**
+     * Checks if given file is at src/{test, main}/outterDirectory/filename exists or not.
+     * @param filename to search
+     * @param outterDirectory to append after test or main
+     * @return Location of searched file
+     */
+    static URI checkSrcTestAndMainResources(String filename, String outterDirectory) {
+        final Path testPath = Paths.get("src", "test", outterDirectory, filename);
+        Path finalPath;
+        if ((finalPath = exists(testPath)) != null) {
+            return finalPath.toUri();
+        } else {
+            final Path mainPath = Paths.get("src", "main", outterDirectory, filename);
+            if ((finalPath = exists(mainPath)) != null) {
+                return finalPath.toUri();
+            }
+        }
+
+        return null;
+    }
+
+    private static Path exists(Path fullpath) {
+        final Path ymlPath = fullpath.resolveSibling(fullpath.getFileName() + ".yml");
+        if (Files.exists(ymlPath)) {
+            return ymlPath;
+        } else {
+            final Path yamlPath = fullpath.resolveSibling(fullpath.getFileName() + ".yaml");
+            if (Files.exists(yamlPath)) {
+                return yamlPath;
+            }
+        }
+        
+        return null;
```

Comment:
Originally I had with optional but at the end since are internal methods I was not really sure about it. I think that Optional should be used for public methods, maybe for private ones has no sense at all.